uint8[] jpg_data
